rfcs/0007-consistent-vehicle-ids.md

```diff
@@ -0,0 +1,186 @@
+- Feature Name: `consistent-vehicle-ids`
+- Start Date: 2021-09-14
+- RFC PR: [mbta/technology-docs#7](https://github.com/mbta/technology-docs/pull/7)
+- Asana task: [changing vehicle ids to make them consistent?](https://app.asana.com/0/116562505129567/1200545747038396)
+- Status: Proposed
+
+# Summary
+[summary]: #summary
+
+Internal MBTA application publish vehicle IDs with a consistent prefix so as to avoid overlaps.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the MBTA runs vehicles on 5 different modes:
+
+- Bus
+- Commuter Rail
+- Ferry
+- Heavy Rail (Red, Orange, and Blue Lines)
+- Light Rail (Green Line and Mattapan)
+
+Within and between these modes, the identifiers of the vehicle (printed on
+the outside) can overlap, causing confusion in our real-time data. By
+creating consistent prefixes for each mode, we can ensure that the
+identifiers used in our real-time data are unique across all modes.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+<!--
+Explain the proposal as if it was already implemented and you were teaching it to a new developer
+that just joined the team. That generally means:
+
+- Introducing new named concepts.
+- Explaining the feature largely in terms of examples.
+- Explaining how programmers should *think* about the feature, and how it should impact the way they
+  work on this project. It should explain the impact as concretely as possible.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to senior developers and to junior
+  developers. -->
+
+Our vehicle information is published in our GTFS-RT VehiclePositions file. There are three fields which currently contain some vehicle information:
+
+| Field | Description | Values |
+| --- | --- | --- |
+| `id` | Internal system of identification for the vehicle. Should be unique to the vehicle | Heavy rail: the line the car is on plus a hex code that is unique for the consist<br>Light rail: "G-" plus a decimal code that is unique for the consist<br>Commuter Rail: the number of the control coach<br>Bus: the letter "y" plus the number of the bus |
+| `label` | a user visible label - for example the name of a train | Heavy rail: the number of the lead car<br>Light rail: the numbers of each car in the consist, separated by `-`<br>Commuter Rail: the number of the control coach<br>Bus: the number of the bus |
```

Comment:
nitpick: For Commuter Rail, when a locomotive is at the front of the train, it'd be the number of the "locomotive" rather than "control coach", right?

Reply:
IIRC the label is associated with the GPS tracking unit, and there's only one of those per train, which is always in the control coach, regardless of which direction it's heading.